include/powsybl/iidm/GenerationAdder.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_GENERATIONADDER_HPP
+#define POWSYBL_IIDM_GENERATIONADDER_HPP
+
+#include <functional>
+
+#include <powsybl/stdcxx/math.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+class DanglingLineAdder;
+
+namespace dangling_line_views {
+
+class GenerationAdder {
+public:
+    explicit GenerationAdder(DanglingLineAdder& adder);
+
+    DanglingLineAdder& add();
+
+    GenerationAdder& setMaxP(double maxP);
+
+    GenerationAdder& setMinP(double minP);
+
+    GenerationAdder& setTargetP(double targetP);
+
+    GenerationAdder& setTargetQ(double targetQ);
+
+    GenerationAdder& setTargetV(double targetV);
+
+    GenerationAdder& setVoltageRegulationOn(bool voltageRegulationOn);
+
+private:
+    friend class iidm::DanglingLineAdder;
+
+private:
+    std::reference_wrapper<DanglingLineAdder> m_danglingLineAdder;
```

Comment:
To be discussed : use a reference_wrapper here to make GenerationAdder assignable which is needed in DanglingLineAdder::newGeneration() to avoid using pointers in GenerationAdder because this class should be copiable/affectable (adders are returned by copy)